pkg/api/imagereferencemutators/internalversion/pods.go

```diff
@@ -0,0 +1,421 @@
+	GetContainerByName(name string) (ContainerMutator, bool)
+	Path() *field.Path
+}
+
+// GetPodSpecReferenceMutator returns a mutator for the provided object, or an error if no
+// such mutator is defined.
+func GetPodSpecReferenceMutator(obj runtime.Object) (PodSpecReferenceMutator, error) {
+	if spec, path, err := GetPodSpec(obj); err == nil {
+		return &podSpecMutator{spec: spec, path: path}, nil
+	}
+	if spec, path, err := GetPodSpecV1(obj); err == nil {
+		return &podSpecV1Mutator{spec: spec, path: path}, nil
+	}
+	return nil, errNoImageMutator
+}
+
+var errNoPodSpec = fmt.Errorf("No PodSpec available for this object")
+
+// GetPodSpec returns a mutable pod spec out of the provided object, including a field path
+// to the field in the object, or an error if the object does not contain a pod spec.
+// This only returns internal objects.
+func GetPodSpec(obj runtime.Object) (*kapi.PodSpec, *field.Path, error) {
+	switch r := obj.(type) {
+	case *kapi.Pod:
+		return &r.Spec, field.NewPath("spec"), nil
+	case *kapi.PodTemplate:
+		return &r.Template.Spec, field.NewPath("template", "spec"), nil
+	case *kapi.ReplicationController:
+		if r.Spec.Template != nil {
+			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+		}
+	case *extensions.DaemonSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensions.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *extensions.ReplicaSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *batch.Job:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *batch.CronJob:
+		return &r.Spec.JobTemplate.Spec.Template.Spec, field.NewPath("spec", "jobTemplate", "spec", "template", "spec"), nil
+	case *batch.JobTemplate:
+		return &r.Template.Spec.Template.Spec, field.NewPath("template", "spec", "template", "spec"), nil
+	case *apps.StatefulSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicySubjectReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicySelfSubjectReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *securityapi.PodSecurityPolicyReview:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsapi.DeploymentConfig:
+		if r.Spec.Template != nil {
+			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+		}
+	}
+	return nil, nil, errNoPodSpec
+}
+
+// GetPodSpecV1 returns a mutable pod spec out of the provided object, including a field path
+// to the field in the object, or an error if the object does not contain a pod spec.
+// This only returns pod specs for v1 compatible objects.
+func GetPodSpecV1(obj runtime.Object) (*kapiv1.PodSpec, *field.Path, error) {
```

Comment:
That's wrong, if you're creating one package with externals and another with internals this should live in externals.

Reply:
@soltysh mind a follow up? this was pre-existing and I don't want to deal with ripples of changing this at this point.